typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,28 @@
+const $tabContainer = document.querySelector('.tab-container');
+const $tabList = document.querySelectorAll('.tab');
+const $viewList = document.querySelectorAll('.view');
+if (!$tabContainer || !$tabList || !$viewList) {
+  throw new Error('Query failed');
+}
+
+$tabContainer.addEventListener('click', (event: Event) => {
+  const $eventTarget = event.target as HTMLDivElement;
+  let $view: unknown = '';
+  if ($eventTarget.matches('.tab')) {
+    $tabList.forEach((div) => {
```

Comment:
I would rename the parameter. `div` works, but what if we edit our HTML and they are no longer a `div`?

A better name would probably be something like `tab`